File: src/RESR.Frontend/src/shared/types/reaction.ts
export type ReactionName = "like" | "love" | "dislike";

export interface ReactionUser {
  idUser: number;
  username: string;
  firstName: string;
}

export interface ResourceReaction {
  idReaction: number;
  name: ReactionName;
  idResource: number;
  idUser: number;
  user: ReactionUser;
}

export interface CreateReactionRequest {
  name: ReactionName;
}

export interface UpdateReactionRequest {
  name: ReactionName;
}

File: src/RESR.Models/Reactions/Reaction.cs
namespace RESR.Models.Reactions;

public sealed class Reaction
{
    public int IdReaction { get; set; }
    public string Name { get; set; } = string.Empty;
    public int IdResource { get; set; }
    public int IdUser { get; set; }
    public string Username { get; set; } = string.Empty;
    public string FirstName { get; set; } = string.Empty;
}
